### DISCUSSION TITLE
Insert text at cursor via custom command

### DISCUSSION THREAD

**Discussion by ole** [notaapp/talk#205] [Questions] [answered]
Is it possible to write a custom command that inserts text at the current cursor position?

I wanted to write a command to insert the current date:

```sh
#!/bin/bash
# @parameters file
# I don't want to use the file param, but it doesn't hurt

# Prints the current date in YYYY-MM-DD
date -u +"%Y-%m-%d"
```

I sort of expected the output of my command to stdout to be inserted at the cursor position, but that doesn't seem to be happening.

I know the command can receive the current file path, but (a) rewriting the file by hand is complicated and (b) the command doesn't know the current cursor position (or selection range).

Thanks!

**Comment by astoilkov** [comment, marked as answer]
> I sort of expected the output of my command to stdout to be inserted at the cursor position, but that doesn't seem to be happening.

You are not the first to expect that behavior. I'm currently working on writing better docs for the extensibility and in parallel on extensibility improvements. I will try to implement this as it makes a lot of sense. Will keep you posted.

> I know the command can receive the current file path, but (a) rewriting the file by hand is complicated and (b) the command doesn't know the current cursor position (or selection range).

I agree, this will be more complicated. However, I will be a nice addition to support passing the selection.

> I wanted to write a command to insert the current date:

We're also planning better support for dates. Quick insertions, parsing and highlighting them, daily notes, a calendar (only if daily notes feature is enabled), etc.

---

BTW, have you tried using third-party apps for that? The most common apps people use are:
- [TextExpander – Productivity App | Text Expander](https://textexpander.com/) is probably the most popular way to solve this problem. Writing a specific phrase will insert the associated snippet for this phrase. You can have a snippet like `;today` that will insert the today's date.
- [Alfred](https://www.alfredapp.com/). If you are already a paying user of Alfred you can use their [Snippets](https://www.alfredapp.com/help/features/snippets/) feature to quickly add snippets in any app.
- [Raycast](https://raycast.com/). Use the Clipboard History feature of Raycast. Assign a shortcut to the Clipboard History to access it with a single shortcut and then pin items(using `cmd-k`) that will act as snippets. The bonus is that you will get clipboard history as well.

The benefit of using a third-party app is that it's a universal solution — once you learn it, you can use it in every other app.

**Reply by ole** [reply]
Thanks for all the info.

> BTW, have you tried using third-party apps for that?

Good suggestions, thank you. It’s a tradeoff, as always. Having the command in Obsidian means I can use it on _in Obsidian_ on all devices (Mac and iOS for me). I wouldn't get that with a Mac-wide third-party solution.
